drivers/sensor/ina23x/ina237.c

```diff
@@ -165,6 +183,14 @@ static int ina237_read_data(const struct device *dev)
 		}
 	}
 
+	if ((data->chan == SENSOR_CHAN_ALL) || (data->chan == SENSOR_CHAN_VSHUNT)) {
```

Comment:
> Is there a real use case for reading the shunt voltage?

Very true, it is just for debug at this point, although I have considered doing the math in the micro instead just as you mentioned later.  I'll remove it from the SENSOR_CHAN_ALL group.  Based upon the extra cached value, did you want me to actually remove this completely?

> TBH, until someone ports Zephyr to the 8051, all the conversion and math stuff this chip does is obsolete. The most efficient driver would just read shunt voltage and voltage, ignore the cal stuff totally, and do all the math on the CPU

Yes, I agree with you on this part after going through everything.  I also looked at doing a bulk read of all of the registers so at least there is only one I2C transaction, but that is not supported by the chip :(

Reply:
Removed it from the ALL_CHANNELS list in the next patchset.

Reply:
@xyzzy42 Please take another look and resolve this thread if you are happy with removing the shunt voltage from ALL_CHANNELS.

Reply:
What I think doesn't really matter, but I'd probably make it optional via a menuconfig entry.  I wouldn't want to enable it by default, since it has a continuous runtime cost with zero benefit to virtually all users.  But it could be useful for debugging or some unusual use case like non-constant shunt resistance value or using the INA237 as a mV meter.  And I do like to expose every feature I can.

Having it there, but not in the ALL_CHANNELS list seems like it would be confusing.  I don't think there's a precedent for that and it's contrary to the sensor docs.

Reply:
@xyzzy42 I have added the KConfig and kept the current shunt in the ALL_CHANNEL group.  Please take another look when you have a chance.